az-digital/az_quickstart
# Closes #1363 Use navbar_offcanvas_attributes set in .theme file.

Remove navbar off-canvas class logic from theme.
Use navbar_offcanvas_attributes var set in .theme file.
Add PHPUnit tests to check functionality and protect from regressions.

## Related Issue
Closes #1363 

## How to test this:
On install check that the appropriate classes and id exist.
By default, there is a block in the navbar region and a block in the navbar off_canvas region
When this is the case, there should be an element on the page that can be selected with this selector
`#navbar-top.has-navbar-region.has-navbar-offcanvas-region`

If you change the theme settings for arizona barrio to disable navbar off canvas there should no longer be an ID of `navbar-top` or any of these classes on any page using the arizona barrio theme.
```
has-navbar-region
has-navbar-offcanvas-region
no-navbar-region
no-navbar-offcanvas-region
```
if you re-enable navbar off canvas in the arizona barrio theme, the classes should reappear.

If you remove all blocks from the navbar region, you should now see an element that can be selected with this
`#navbar-top.no-navbar-region.has-navbar-offcanvas-region`

If you remove all blocks from the navbar off-canvas region, you should now see an element that can be selected with this
`#navbar-top.no-navbar-region.no-navbar-offcanvas-region`

If you add a block back to the navbar region, you should now see an element that can be selected with this
`#navbar-top.has-navbar-region.no-navbar-offcanvas-region`

If you add a block back to the navbar off-canvas region, you should now see an element that can be selected with this
`#navbar-top.has-navbar-region.has-navbar-offcanvas-region`

This is marked as `Breaking or visual change to existing behavior` since it could possibly have an effect on Google Tag Manager, or Google analytics tracking.

If you are tracking anything using the `navbarOffcanvasDemo` ID, you will need to change that to `navbar-top`
If you have custom CSS using the `navbarOffcanvasDemo` ID, you will need to change that to `navbar-top`
If you have custom JS using the `navbarOffcanvasDemo` ID, you will need to change that to `navbar-top`

## Types of changes

**Arizona Quickstart** (install profile, custom modules, custom theme)
- Patch release changes
- Minor release changes

**Drupal core**
- Patch release changes
- Minor release changes

**Drupal contrib projects**
- Patch release changes
- Minor release changes

## Checklist:
